tools/azure-pipelines/build-nightly-dist.yml

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+jobs:
+  - job: ${{parameters.stage_name}}_binary
+    pool:
+      vmImage: 'ubuntu-16.04'
+    container: flink-build-container
+    workspace:
+      clean: all
+    steps:
+      - task: Cache@2
+        inputs:
+          key: $(CACHE_KEY)
+          restoreKeys: $(CACHE_FALLBACK_KEY)
+          path: $(MAVEN_CACHE_FOLDER)
+        displayName: Cache Maven local repo
+        continueOnError: true
+      # Create binary release
+      - script: |
```

Comment:
is it a hard requirement that the script is the first item? If not, then having the displayName as the first item would eliminate the requirement of needing a comment to explain what it does.

Reply:
I don't think there's anything we can do about it.

Reply:
what we could maybe do is specify `task: CmdLine@2`, have the display name below, then the script block?
I think we can make these files much more readable if the display name is closer to the start of the block.

Reply:
Yep, that works. Once I have validated that the S3 upload works, I'll update the PR.